src/index.ts

```diff
@@ -47,7 +49,8 @@ function validate(options: Options) {
 async function metadataAccessor<T>(
   type: string,
   options?: string | Options,
-  noResponseRetries = 3
+  noResponseRetries = 3,
+  fastFail = false
```

Comment:
I see that you aren't changing the default behavior for everyone; just for `isAvailable`. Is there a reason for not changing this for the other paths, e.g. if my library doesn't call `isAvailable` but rather directly tries to lookup a project or instance property. What are the scenarios where the slow-fail approach is desirable for those paths?

Perhaps you are being cautious around it being potentially breaking, but IMO, 'taking too long to fail' was a bug, and we should be changing the default behavior.

Reply:
My goal was to be cautious, since it seemed that `isAvailable` is the culprit biting `@google-cloud/logging`.

In other libraries we've looked at, we don't seem to call `isAvailable` once we see `OOGLE_APPLICATION_CREDENTIALS`; I suspect we could be similarly cautious, where we only lookup metadata if we've confirmed we're in a `GCP` environment.